test/Microsoft.AspNetCore.Routing.Tests/Matching/MatcherConformanceTest.cs

```diff
@@ -21,7 +21,10 @@ internal static (HttpContext httpContext, EndpointFeature feature) CreateContext
             httpContext.Request.Path = path;
             httpContext.RequestServices = CreateServices();
 
-            var feature = new EndpointFeature();
+            var feature = new EndpointFeature
+            {
+                RouteValues = new RouteValueDictionary()
```

Comment:
Is this needed for correctness? I guess I don't see any changes to the middleware, but I hope we don't need to create an empty RVD each time we allocate a feature.

Reply:
+1 to avoiding that.

One hitch is if RoutesValues is exposed on HttpRequest then we probably don't want to return a null collection. In the HttpRequest.RouteValues getter we could check if IRouteValuesFeature.RouteValues is null and assign a value lazily, and return that.